staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go

```diff
@@ -272,3 +279,7 @@ func buildImpersonationRequests(headers http.Header) ([]v1.ObjectReference, erro
 
 	return impersonationRequests, nil
 }
+
+func getForbiddenNamespaces() []string {
+	return []string{"sigs.k8s.io/", "kubernetes.io/", "k8s.io/"}
```

Comment:
This is a breaking change for users setting these namespaces on impersonation requests. We should research those use cases, or feature gate this change.

Another option is to enable users to inject their own forbidden namespaces here.